### PR TITLE
Validating pointer value types

### PR DESCRIPTION
**Pull request by trueqbit**
Adressing #851, this PR introduces validation of bound pointer values against the expected pointer value taken as function arguments.

Static validation tells you when you mix wrong pointer value types, e.g.
```c++
    using carray_pvt = std::integral_constant<const char*, carray_pvt_name>;
    using ecode_pvt = std::integral_constant<const char*, ecode_pvt_name>;
    using ecode_arg_t = pointer_arg<std::error_code, ecode_pvt>;

    struct equal_error_code_fn {
        bool operator()(ecode_arg_t pv1, ecode_arg_t pv2) const {
              // ...
              return false;
        }

        static constexpr const char* name() {
            return "equal_error_code";
        }
    };

    // ...

    auto node = func<equal_error_code_fn>(1,
        bindable_pointer<carray_pvt>(make_unique<error_code>()));
```
Instantiating the `function_call` node spits out something almost readable like:

> 1>sqlite_orm\dev\function.h(204,27): error C2338: Pointer value types of I-th argument do not match
> 1>sqlite_orm\dev\function.h(219): message : see reference to function template instantiation 'bool sqlite_orm::internal::assert_same_pointer_type<1,& ecode_pvt_name,& sqlite_orm::carray_pvt_name>(void)' being compiled
> 
> 1>sqlite_orm\dev\function.h(209,27): error C2338: Expected a pointer value for I-th argument
> 1>sqlite_orm\dev\function.h(214): message : see reference to function template instantiation 'bool sqlite_orm::internal::expected_pointer_value<0,FnArg,CallArg>(void)' being compiled
> 1>        with
> 1>        [
> 1>            FnArg=sqlite_orm::pointer_arg<std::error_code,ecode_pvt>,
> 1>            CallArg=int
> 1>        ]
>

### CODE REVIEW

File: dev/start_macros.h [hunk 1]
```diff
@@ -22,11 +22,15 @@ __pragma(push_macro("min"))
 #define SQLITE_ORM_NOTHROW_ALIASES_SUPPORTED
 #define SQLITE_ORM_INLINE_VAR inline
 #if __cplusplus >= 202002L  // use of C++20 or higher
+#define SQLITE_ORM_CONSTEVAL consteval
 #define SQLITE_ORM_NOUNIQUEADDRESS [[no_unique_address]]
 #else
+#define SQLITE_ORM_CONSTEVAL
```
Comment (trueqbit) [inline dev/start_macros.h:28]: Need to deduplicate.

**Comment by trueqbit** [comment]
@fnc12 Pinging you just in case you didn't see it...

**Comment by fnc12** [comment]
merged. Thanks @trueqbit

**Comment by trueqbit** [comment]
@fnc12 Hah, you were a bit quick this time :)

**Pull request merged**